stereo_image_proc/src/nodelets/point_cloud2.cpp

```diff
@@ -231,18 +217,10 @@ void PointCloud2Nodelet::imageCb(const ImageConstPtr& l_image_msg,
                                   l_image_msg->step);
     for (int v = 0; v < mat.rows; ++v)
     {
-      for (int u = 0; u < mat.cols; ++u, offset += STEP)
+      for (int u = 0; u < mat.cols; ++u, ++iter_r, ++iter_g, ++iter_b)
       {
-        if (isValidPoint(mat(v,u)))
-        {
-          uint8_t g = color(v,u);
-          int32_t rgb = (g << 16) | (g << 8) | g;
-          memcpy (&points_msg->data[offset + 12], &rgb, sizeof (int32_t));
-        }
-        else
-        {
-          memcpy (&points_msg->data[offset + 12], &bad_point, sizeof (float));
-        }
+        uint8_t g = color(v,u);
+        *iter_r = *iter_g = *iter_b = g;
```

Comment:
You mean the behavior of xyz copying I guess, I did not change behavior.
It is around L196, please check it.
https://github.com/ros-perception/image_pipeline/pull/115/files#diff-7ad2f03337d702facd01ce7e3b65a7e0R196

This section is color information copying,
but, yes, I changed the behavior.

Before, if the point is invalid, then `NaN(float)` will be copied into `rgb(uint8_t[3] + 1 byte)` and it has no mean in `uint8_t[3]`. It is strange I think, so after, I changed all color information will be copied whether the point is valid or not.

And I'd like to change behavior to copy color if the point is invalid,
because we can reconstruct color image from only point cloud like `depth_image_proc`.